### PR TITLE
Add a configurable time-out to onchain fee provider requests

### PR DESCRIPTION
**Pull request by sstone**
We configure a timeout of 5 seconds, applicable to all fee providers. If a provider times out we switch to the next one in our list.
Our mobile app needs a feerate to start properly and currently waits too long when a fee provider is online but very slow to respond.

### CODE REVIEW

File: eclair-core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -31,6 +31,7 @@ eclair {
 
   min-feerate = 3 // minimum feerate in satoshis per byte
   smooth-feerate-window = 6 // 1 = no smoothing
+  feerate-provider-timeout = 1 second // max time we'll wait for answers from a fee provider before we fallback to the next one
```
Comment (pm47) [inline eclair-core/src/main/resources/reference.conf]: Timeout is really `1 second`, not `5 seconds`.

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/BitgoFeeProvider.scala [hunk 2]
```diff
@@ -23,9 +23,10 @@ import org.json4s.DefaultFormats
 import org.json4s.JsonAST.{JInt, JValue}
 import org.json4s.jackson.Serialization
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
-class BitgoFeeProvider(chainHash: ByteVector32)(implicit http: SttpBackend[Future, Nothing], ec: ExecutionContext) extends FeeProvider {
+class BitgoFeeProvider(chainHash: ByteVector32, readTimeOut: Duration = 5 seconds)(implicit http: SttpBackend[Future, Nothing], ec: ExecutionContext) extends FeeProvider {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/BitgoFeeProvider.scala]: Default values can be error-prone, as it is the case here.

File: eclair-core/src/main/resources/reference.conf (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/resources/reference.conf]: I think we should set 5 seconds here and no default value in the object constructors

Comment (sstone) [reply eclair-core/src/main/resources/reference.conf]: Agreed.

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/BitgoFeeProvider.scala (hunk 2 above)
Comment (sstone) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/BitgoFeeProvider.scala]: Agreed, I've removed them

**Comment by sstone** [comment]
I've added timeout tests (with a timeout of 1 ms to make sure it always fails) to the BigGo and Earn fee provider tests.

**Pull request merged**